tests/PHPStan/Analyser/data/bug-6842.php

```diff
@@ -16,6 +16,7 @@ public function getScheduledEvents(
 		\DateTimeInterface $startDate,
 		\DateTimeInterface $endDate
 	): \Iterator {
+		/** @var \DateInterval $interval */
```

Comment:
a DynamicReturnTypeExtension should be implemented, to make sure this case here does not require this inline `@var` to realize it cannot be `false`

similar to https://github.com/phpstan/phpstan-src/blob/1.9.x/src/Type/Php/DateTimeDynamicReturnTypeExtension.php which is handling the constant string argument